lib/entities/resource-type.ts

```diff
@@ -0,0 +1,140 @@
+import type {
+  BasicMetaSysProps,
+  DefaultElements,
+  GetResourceTypeParams,
+  MakeRequest,
+  SysLink,
+} from '../common-types'
+import { toPlainObject, freezeSys } from 'contentful-sdk-core'
+import copy from 'fast-copy'
+import enhanceWithMethods from '../enhance-with-methods'
+
+export type ResourceTypeProps = {
+  /**
+   * System metadata
+   */
+  sys: Omit<BasicMetaSysProps, 'version'> & {
+    appDefinition: SysLink
+    organization: SysLink
+    resourceProvider: SysLink
+  }
+  /**
+   * Resource Provider name, TODO
+   */
+  name: string
```

Comment:
The resource provider currently doesn’t have a `name` property. At the very least it should be optional but I’d prefer to leave it out completely.

Reply:
You’re right, I got confused by it saying "provider name".